### PR TITLE
feat(java 21)!: upgrade java version to 21

### PR DESCRIPTION
**Pull request by kenluluuuluuuuu**
- Upgrading java version from 11 to 21
- Updating styx-build base image
- Removing deprecated JVM flags
- Adding org.bouncycastle:bcpkix-jdk18on for resolving self signed certificate issue on tests

### CODE REVIEW

File: docs/developer-guide.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ In this guide we will walk through how to build applications on top of Styx.
 
 ## 1.2 System Requirements
 
-Building Styx requires Java 11.
+Building Styx requires Java 21.
 
 The build system requires Apache Maven. The Styx team uses Maven version 3.2.1
```
Comment (NersesAM) [inline docs/developer-guide.md:13]: maybe just generic latest version of maven 3?

File: docker/styx-image/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG TAG=11-jdk
+ARG TAG=21-jdk
 FROM openjdk:${TAG}
```
Comment (NersesAM) [inline docker/styx-image/Dockerfile:2]: should we switch to Temurin here?

File: pom.xml [hunk 3]
```diff
@@ -872,8 +880,8 @@
                   <version>[3.2.1,)</version>
```
[the remark this answers is not in the document]
Comment (kenluluuuluuuuu) [reply pom.xml:880]: I've done it in https://github.com/ExpediaGroup/styx/pull/848/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R875

File: docs/developer-guide.md (hunk 1 above)
Comment (kenluluuuluuuuu) [reply docs/developer-guide.md:13]: Changed it in #848

File: docker/styx-image/Dockerfile (hunk 2 above)
Comment (kenluluuuluuuuu) [reply docker/styx-image/Dockerfile:2]: That's right. Let me change it

Comment (kenluluuuluuuuu) [reply docker/styx-image/Dockerfile]: done

**Review by ThaerAlAjlouni — APPROVED** [review]

**Review by zckoh — APPROVED** [review]

**Pull request merged**